eternalcore-core/src/main/java/com/eternalcode/core/feature/jail/PrisonerServiceImpl.java

```diff
@@ -0,0 +1,158 @@
+package com.eternalcode.core.feature.jail;
+
+import com.eternalcode.core.feature.jail.event.JailDetainEvent;
+import com.eternalcode.core.feature.jail.event.JailReleaseEvent;
+import com.eternalcode.core.feature.spawn.SpawnService;
+import com.eternalcode.core.feature.teleport.TeleportService;
+import com.eternalcode.core.injector.annotations.Inject;
+import com.eternalcode.core.injector.annotations.component.Service;
+import com.eternalcode.core.util.DurationUtil;
+
+import org.bukkit.Server;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+import javax.annotation.Nullable;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Service
+public class PrisonerServiceImpl implements PrisonerService {
+
+    private final Map<UUID, Prisoner> jailedPlayers = new ConcurrentHashMap<>();
```

Comment:
Why this is `ConcurrentHashMap`? I didn't notice that you used multithreading if I'm wrong... correct me, but currently it suggests changing to a regular HashMap.